lldb/source/Expression/DiagnosticManager.cpp

```diff
@@ -65,6 +65,23 @@ std::string DiagnosticManager::GetString(char separator) {
   return ret;
 }
 
+Status DiagnosticManager::GetAsStatus(lldb::ExpressionResults result) {
+  llvm::SmallVector<Status::Detail, 0> details;
+  details.reserve(m_diagnostics.size());
+  for (const auto &diagnostic : m_diagnostics)
+    details.push_back(diagnostic->GetDetail());
+  return Status::FromExpressionErrorDetails(result, std::move(details));
+}
+
+void DiagnosticManager::AddDiagnostic(llvm::StringRef message,
+                                      lldb::Severity severity,
+                                      DiagnosticOrigin origin,
+                                      uint32_t compiler_id) {
+  m_diagnostics.emplace_back(std::make_unique<Diagnostic>(
+      origin, compiler_id,
+      Status::Detail{{}, severity, message.str(), message.str()}));
```

Comment:
Looks like in this case, `Status::Detail::message` is the same as `Status::Detail::rendered`. Is that really what we want ? If so, do we need both variables then ?

Reply:
See https://github.com/llvm/llvm-project/pull/106470

Reply:
Correct. For manual LLDB-originated diagnostics the two are identical. Generally the "rendered" version is a fallback that exists if the user interface cannot render the diagnostic itself. In the case of the expression evaluator the big difference is that the rendered version repeats the expressions string inside of the diagnostic.